lib/Checks/TrailingWhitespaceCheck.vala

```diff
@@ -28,17 +28,17 @@ public class ValaLint.Checks.TrailingWhitespaceCheck : Check {
         return _("Checks for whitespaces at the end of lines");
     }
 
-    public override bool check_line (Gee.ArrayList<FormatMistake?> mistake_list, int line_index, string line) {
-        string clean_line = line.chomp ();
-
-        if (clean_line != line) {
-            string mistake = _("Unexpected whitespace at end of line");
-
-            mistake_list.add ({ this, line_index, clean_line.length, mistake });
-
-            return true;
+    public override void check (Gee.ArrayList<ParseResult? > parse_result, Gee.ArrayList<FormatMistake? > mistake_list) {
+        foreach (ParseResult r in parse_result) {
+            if (r.type == ParseType.Default) {
+                add_regex_mistake (this, """ \n""", "Unexpected whitespace at end of line", r, mistake_list);
```

Comment:
Should probably switch this to `\h` (horizontal whitespace) instead of a literal space character. People could have put all kinds of whitespace at the end of a line.